src/optimizerinterface.jl

```diff
@@ -3,6 +3,27 @@
 #  License, v. 2.0. If a copy of the MPL was not distributed with this
 #  file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+"""
+    setoptimizer(model::Model, factory::Factory)
+
+Sets the optimizer of the model `model` as the optimizers created by the
+factory `factory`. The factory can be created by the [`with_optimizer`](@ref)
+function.
+
+## Examples
+
+The following sets the optimizer of `model` to be
+`IpoptOptimizer(print_level=0)`:
+```julia
+setoptimizer(model, with_optimizer(IpoptOptimizer, print_level=0))
+```
+"""
+function setoptimizer(model::Model, factory::Factory)
```

Comment:
We really need to decide on a convention for underscores. `with_optimizer` and `setoptimizer` are inconsistent.

Reply:
I agree with using underscores, it's a typo :)